tests/framework/interface.go

```diff
@@ -30,6 +30,7 @@ type testRunner interface {
 type Cluster interface {
 	Members() []Member
 	Client() Client
+	ClientWithAuth(userName, password string) Client
```

Comment:
I think we should add Options argument to `Client` instead.

Reply:
I agree. To not break other in-flight tests migration PR, can we defer to a follow-up PR?

Reply:
I think we can do a minimal PR targeted just to fix it it should be ok. Conflicts will still happen, let's just make sure PRs will be quickly merged and conflics are trivial to resolve.

Reply:
SGTM